src/game_api/script.cpp

```diff
@@ -1037,6 +1037,27 @@ SpelunkyScript::ScriptImpl::ScriptImpl(std::string script, std::string file, Sou
         }
         return (uint16_t)0;
     };
+    /// Read input that has been previously stolen with steal_input
+    lua["read_stolen_input"] = [this](int uid)
+    {
+        Player* player = get_entity_ptr(uid)->as<Player>();
+        if (player == nullptr)
+            return (uint16_t)0;
+        ScriptInput* readinput = reinterpret_cast<ScriptInput*>(player->input_ptr);
```

Comment:
So this cast is only valid if the input was actually stolen, you should check if that is the case before using this pointer.

Reply:
dumb question, is there a standardized way to check if the pointer is valid? do i use the IsBadReadPtr method?

Reply:
Not sure if this is the change you were looking for or if there is a way that I can read the pointer in a way to tell if it is valid or not. I ended up adding a boolean flag to tell if the input was stolen